src/Pangraph.hs

```diff
@@ -135,41 +133,37 @@ vertexAttributes = vertexAttributes'
 edgeEndpoints :: Edge -> (Vertex, Vertex)
 edgeEndpoints = endpoints'
 
-{-| Returns the EdgeID if it has one. `Edge`s are given a new `VetexID` when they are passed and retrived from a `Pangraph`-}
+{-| Returns the EdgeID if it has one. 'Edge's are given a new 'EdgeID' when they are passed and retrived from a 'Pangraph'-}
 edgeID :: Edge -> Maybe EdgeID
 edgeID = edgeID'
 
--- | Returns a `VetexID`
+-- | Returns a 'VetexID'
 vertexID :: Vertex -> VertexID
 vertexID = vertexID'
 
 -- * Operations on Edge and Vertex
 
--- | Lookup Attributes by `Key`s
--- | Complexity: /O(n)/
+-- | Lookup Attributes by 'Key's Complexity: /O(n)/
```

Comment:
If the complexity is linear (i.e. quite expensive), why not allow the user to simply get all attributes as a list and arrange the search themselves as they like, potentially via `lookup` like you do?

I would then call this function `vertexAttributes`.